drivers/ethernet/dsa_ksz8794.c

```diff
@@ -0,0 +1,413 @@
+	     tmp != KSZ8794_CHIP_ID0_ID_DEFAULT && timeout > 0; timeout--) {
+		dsa_ksz8794_read_reg(sdev, KSZ8794_CHIP_ID0, &tmp);
+		k_busy_wait(100);
+	}
+
+	if (timeout == 0) {
+		LOG_ERR("KSZ8794: No SPI communication!");
+		return -ENODEV;
+	}
+
+	dsa_ksz8794_read_reg(sdev, KSZ8794_CHIP_ID0, &val[0]);
+	dsa_ksz8794_read_reg(sdev, KSZ8794_CHIP_ID1, &val[1]);
+
+	LOG_DBG("KSZ8794: ID0: 0x%x ID1: 0x%x timeout: %d", val[1], val[0],
+		timeout);
+
+	return 0;
+}
+
+static int dsa_ksz8794_switch_setup(struct dsa_ksz8794_spi *sdev)
+{
+	u8_t tmp, i;
+
+	/* Disable tail tag feature */
+	dsa_ksz8794_read_reg(sdev, KSZ8794_GLOBAL_CTRL10, &tmp);
+	tmp &= ~KSZ8794_GLOBAL_CTRL10_TAIL_TAG_EN;
+	dsa_ksz8794_write_reg(sdev, KSZ8794_GLOBAL_CTRL10, tmp);
+
+	/* Loop through ports */
+	for (i = KSZ8794_PORT1; i <= KSZ8794_PORT3; i++)
+	{
+		/* Enable transmission, reception and switch address learning */
+		dsa_ksz8794_read_reg(sdev, KSZ8794_CTRL2_PORTn(i), &tmp);
+		tmp |= KSZ8794_CTRL2_TRANSMIT_EN;
+		tmp |= KSZ8794_CTRL2_RECEIVE_EN;
+		tmp &= ~KSZ8794_CTRL2_LEARNING_DIS;
+		dsa_ksz8794_write_reg(sdev, KSZ8794_CTRL2_PORTn(i), tmp);
+	}
+
+	dsa_ksz8794_read_reg(sdev, KSZ8794_PORT4_IF_CTRL6, &tmp);
+	LOG_DBG("KSZ8794: CONTROL6: 0x%x port4", tmp);
+
+	dsa_ksz8794_read_reg(sdev, KSZ8794_PORT4_CTRL2, &tmp);
+	LOG_DBG("KSZ8794: CONTROL2: 0x%x port4", tmp);
+
+	return 0;
+}
+
+/* Low level initialization code for DSA PHY */
+int dsa_hw_init(struct device *dev)
+{
+	struct dsa_ksz8794_spi *swspi = &phy_spi;
+
+	/* Time needed for KSZ8794 to completely power up (100ms) */
+	k_busy_wait(100000);
+
+	/* Configure SPI */
+	dsa_ksz8794_spi_setup(swspi, DT_ALIAS_DSA_SPI_CLOCK_FREQUENCY,
+			      0, DT_ALIAS_DSA_SPI_CS_GPIOS_CONTROLLER,
+			      DT_ALIAS_DSA_SPI_CS_GPIOS_PIN);
+
+	/* Soft reset */
+	dsa_ksz8794_soft_reset(swspi);
+
+	/* Setup KSZ8794 */
+	dsa_ksz8794_switch_setup(swspi);
+
+	/* Read ports status */
+	dsa_ksz8794_link_status(swspi);
+
+	swspi->is_init = true;
+
+	return 0;
+}
+
+static void dsa_delayed_work(struct k_work *item)
+{
+	struct dsa_context *context =
+		CONTAINER_OF(item, struct dsa_context, dsa_work);
+	bool link_state;
+	u8_t i;
+
+	for (i = KSZ8794_PORT1; i <= KSZ8794_PORT3; i++) {
+		link_state = dsa_ksz8794_port_link_status(&phy_spi, i);
+		if (link_state && !context->link_up[i]) {
+			LOG_INF("DSA port: %d link UP!", i);
+			net_eth_carrier_on(context->iface_slave[i]);
+		} else if (!link_state && context->link_up[i]) {
+			LOG_INF("DSA port: %d link DOWN!", i);
+			net_eth_carrier_off(context->iface_slave[i]);
+		}
+		context->link_up[i] = link_state;
+	}
+
+	k_delayed_work_submit(&context->dsa_work, DSA_STATUS_TICK_MS);
+}
+
+/*
+ * Info regarding ports shall be parsed from DTS - as it is done in Linux
+ * and moved to ./subsys/net/l2/ethernet/dsa/dsa.c
+ */
+int dsa_port_init(struct device *dev)
+{
+	struct dsa_context *context = dev->driver_data;
+	struct dsa_ksz8794_spi *swspi = &phy_spi;
+
+	if (swspi->is_init)
```

Comment:
you will need to put { }, this is a tiny difference with the linux code style we apply.

Apply this rule everywhere

Reply:
OK. If I may ask - why Zephyr introduced such difference?